code/io/joy-sdl.cpp

```diff
@@ -374,7 +382,7 @@ namespace joystick
 
 	int Joystick::getHatDownCount(int index, int pos, bool reset) {
 		Assertion(index >= 0 && index < numHats(), "Invalid index %d!", index);
-		Assertion(pos >= 0 && pos < HAT_NUM_POS, "Invalid hat position %d!", pos);
+		Assertion(pos >= 0 && pos < JOY_NUM_HAT_POS, "Invalid hat position %d!", pos);
```

Comment:
This restriction should be done in the compatibility layer, not the actual joystick API which supports up to 8 (or 9) hat positions.